src/mx_bluesky/hyperion/external_interaction/agamemnon.py

```
@@ -81,15 +106,46 @@ def get_next_instruction(beamline: str) -> dict:
     return _get_parameters_from_url(AGAMEMNON_URL + f"getnextcollect/{beamline}")
 
 
-def get_pin_type_from_agamemnon(beamline: str) -> PinType:
-    params = get_next_instruction(beamline)
-    return _get_pin_type_from_agamemnon_parameters(params)
+def get_visit_from_agamemnon_parameters(parameters: dict) -> str:
+    prefix = parameters.get("prefix")
+    if not prefix:
+        raise KeyError("Failed to get prefix from Agamemnon")
+
+    match = re.match(MX_GENERAL_ROOT_REGEX, prefix)
+
+    if match:
+        return match.group("visit")
+
+    raise ValueError(
+        f"Agamemnon prefix '{prefix}' does not match MX-General root structure"
+    )
+
+
+def compare_params(agamemnon_params, load_centre_collect_params):
+    # Blank parameters
+    parameters = AgamemnonLoadCentreCollect()
+    try:
+        # Populate parameters from Agamemnon
+        visit = get_visit_from_agamemnon_parameters(agamemnon_params)
+        parameters.visit = visit
+
+        # Log differences against GDA populated parameters
+        differences = []
+        for key, value in load_centre_collect_params.__dict__.items():
+            if value != getattr(parameters, key, None):
+                differences.append(key)
+        LOGGER.info(f"Differences found for: {differences}")
+    except (ValueError, KeyError) as e:
+        LOGGER.warning(f"Failed to compare parameters: {e}")
+    except Exception as e:
+        LOGGER.warning(f"Unexpected error occurred. Failed to compare parameters: {e}")
```

Not sure about this error handling, but wanted to swallow the errors here instead of in `update_params_from_agamemnon`

---

Yh, i think swallowing them here is fine :)